src/Tinfoil/Data/Key.hs

```diff
@@ -32,7 +32,13 @@ newtype SymmetricKey =
     unSymmetricKey :: ByteString
   } deriving (Eq, Generic)
 
-instance NFData SymmetricKey where rnf = genericRnf
+instance NFData SymmetricKey where
+  rnf = genericRnf
+
+-- | We use fixed valid Haskell tokens here to help prevent accidents such as
+-- inadvertantly logging secret keys.
+instance Show SymmetricKey where
+  show _ = "SymmetricKey Redacted"
```

Comment:
If we are writing these manually, we should be using `showsPrec` I believe /cc @jystic